pkg/applicationmanager/controllers/applicationregistration.go

```diff
@@ -0,0 +1,82 @@
+package controllers
+
+import (
+	"strings"
+
+	stork_api "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/portworx/sched-ops/k8s/stork"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	// IBMApp registration name
+	IBMApp = "ibm-app-reg"
+	// CouchBaseApp registration name
+	CouchBaseApp = "couchbase-app-reg"
+	// RedisClusterApp registration name
+	RedisClusterApp = "redis-app-reg"
+	// CassandraApp registration name
+	CassandraApp = "cassandra-app-reg"
+)
+
+func getSupportedCRD() map[string][]string {
+	// Default CRD registration of format,
+	// {kind, crdName, suspendPath,suspendType(bool/int)}
+	defCRD := make(map[string][]string)
+	// IBM CRD's
+	defCRD[IBMApp] = []string{"IBPCA,ibpcas.ibp.com",
```

Comment:
Using csv is very error-prone. Can you define the structs here? It's alright if it is more lines of code